tests/integration/test_tf_symbolic.py

```diff
@@ -90,20 +91,29 @@ def test_eng_run_eval_false_measurements_are_tensors(self, setup_eng):
         val = q[0].val
         assert isinstance(val, tf.Tensor)
 
+    @pytest.mark.xfail(reason="Symbolic parameter system only works with TensorFlow 2 objects", raises=AttributeError, strict=True)
     def test_eng_run_with_session_and_feed_dict(self, setup_eng, batch_size, cutoff, tol):
         """Tests whether passing a tf Session and feed_dict
         through `eng.run` leads to proper numerical simulation."""
-        a = tf.Variable(0.5)
+        tf_a = tf.Variable(0.5)
+        tf_phi = tf.Variable(0.7)
         sess = tf.Session()
         sess.run(tf.global_variables_initializer())
-        tf_params = {'session': sess, 'feed_dict': {a: 0.0}}
+        tf_params = {'session': sess, 'feed_dict': {tf_a: 0.0, tf_phi: 1.0}}
 
+        # NOTE this test worked before because the phase was a number, we only needed working tf.Tensor arithmetic in that case.
+        # Now also the pf.exp function has to work on tf.Tensors.
         eng, prog = setup_eng(1)
-        x = prog.params('a')  # free parameter
+        x, y = prog.params('a', 'phi')  # free parameters
         with prog.context as q:
-            Dgate(x) | q
+            Dgate(x, y) | q
 
-        state = eng.run(prog, args={'a': a}, run_options=tf_params).state
+        # use TF to evaluate symbolic parameter expressions
+        strawberryfields.parameters.par_evaluate.lambdify_printer = ['tensorflow']
```

Comment:
Here we change the symbolic parameter lambdify printer to one that can handle TF objects, and after running the program we change it back. I'm not 100% sure if it's a good idea to leave this to the user though.

Another possibility would be to add a check in `Engine.run` to see if the `'session'` run_option is present, which I think is required to use TF objects as parameters, and switch to the tensorflow printer just for that run if this is the case.